apps/fipsinstall.c

```diff
@@ -418,3 +443,53 @@ int fipsinstall_main(int argc, char **argv)
     free_config_and_unload(conf);
     return ret;
 }
+
+static int self_test_events(OSSL_PARAM *params)
+{
+    OSSL_PARAM *p = NULL;
+    const char *phase = NULL, *type= NULL, *desc = NULL;
+    int ret = 0;
+
+    p = OSSL_PARAM_locate(params, OSSL_PROV_PARAM_SELF_TEST_PHASE);
+    if (p == NULL || p->data_type != OSSL_PARAM_UTF8_STRING)
```

Comment:
It might not be relevant here but calling `OSSL_PARAM_get_utf8_string` might be a better option.  It mallocs the string storage space if the passed pointer is null.  This avoids string life times and size concerns and does some of the checking for you.

It doesn't mean deallocation at the end though.  The _PTR types are designed for long lived things, the base string types are not.

Reply:
The strings are all static so it should be ok how it is..